packages/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -4,226 +4,209 @@ import { CheckboxState } from './Checkbox.types';
 
 export const checkboxClassName = 'fui-Checkbox';
 
+// TODO replace these spacing constants with theme values once they're on the theme
+const spacingHorizontalS = '8px';
+const spacingHorizontalM = '12px';
+
+// CSS variables for the indicator colors.
+// These let the colors be picked based on the :hover, etc. states on the root
+const indicatorColor = '--fui-Checkbox-indicator-color';
+const indicatorBorderColor = '--fui-Checkbox-indicator-borderColor';
+const indicatorBackgroundColor = '--fui-Checkbox-indicator-backgroundColor';
+const indicatorSize = '--fui-Checkbox-indicator-size';
+
 /**
  * Styles for the root slot
  */
-const useStyles = makeStyles({
-  root: theme => ({
-    display: 'inline-flex',
+const useRootStyles = makeStyles({
+  base: {
     position: 'relative',
-    alignSelf: 'flex-start',
-    alignItems: 'center',
-    ...shorthands.padding('4px'),
-    userSelect: 'none',
+    display: 'inline-flex',
+    flexDirection: 'row',
+    justifyContent: 'start',
     cursor: 'pointer',
-  }),
+    columnGap: spacingHorizontalM,
+    ...shorthands.padding(spacingHorizontalS),
+  },
 
-  disabled: theme => ({
-    color: theme.colorNeutralForegroundDisabled,
-    cursor: 'default',
+  medium: {
+    [indicatorSize]: '16px',
+  },
 
-    [`& .${checkboxClassName}-indicator`]: {
-      borderColor: theme.colorNeutralStrokeDisabled,
-      color: theme.colorNeutralForegroundDisabled,
-      backgroundColor: theme.colorNeutralBackground1,
-    },
+  large: {
+    [indicatorSize]: '20px',
+  },
 
-    ':hover': {
-      [`& .${checkboxClassName}-indicator`]: {
-        borderColor: theme.colorNeutralStrokeDisabled,
-        color: theme.colorNeutralForegroundDisabled,
-        backgroundColor: theme.colorNeutralBackground1,
-      },
-    },
+  labelBefore: {
+    flexDirection: 'row-reverse',
```

Comment:
I don't think this is an issue, but it might be worth documenting or calling out that `labelPosition: before` will result in the DOM order not matching visual order for label vs. checkbox. I can foresee a11y testers logging bugs, and we should have documentation that says "we thought about this, it's on purpose, and not actually a problem".

Doesn't need to be in this PR necessarily, just something we should include somewhere.

Reply:
Button has the layout of the icon before/after as part of the renderButton which would preserve the dom order.

Reply:
Sounds like changing the DOM order is a better approach; I'll do that instead.